pkg/segment/utils/number_test.go

```diff
@@ -0,0 +1,402 @@
+package utils
+
+import (
+	"math"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+// Test_isInvalid tests if the number is invalid. It should return invalidType.
+func Test_isInvalid(t *testing.T) {
+	n := Number{}
+	n.SetInvalidType()
+	assert.True(t, n.ntype() == invalidType)
+}
+
+// Test_invalidTypeReturnsErrorForInt64AndFloat64 tests converting invalidType to Int64 and Float64.
+func Test_invalidTypeReturnsErrorForInt64AndFloat64(t *testing.T) {
+	n := Number{}
+	n.SetInvalidType()
+	if val, err := n.Int64(); err == nil || val != 0 {
+		assert.Fail(t, "Expected 0, got %v with error %v", val, err)
+	}
+	if val, err := n.Float64(); err == nil || val != 0 {
+		assert.Fail(t, "Expected 0, got %v with error %v", val, err)
+	}
+}
+
+// Test_isBackfill tests if the number is a backfill. It should return 0 for both Int64 and Float64.
+func Test_isBackfill(t *testing.T) {
+
+	n := Number{}
+	n.SetBackfillType()
+	assert.True(t, n.ntype() == backfillType)
+
+	intVal, err := n.Int64()
+	assert.True(t, intVal == 0, nil)
+	assert.True(t, err == nil)
+
+	floatVal, err := n.Float64()
+	assert.True(t, floatVal == 0, nil)
+	assert.True(t, err == nil)
+}
+
+// Test_isInt64 tests if the number is an int64. It should return set value for Int64 and an error for Float64.
+func Test_isInt64(t *testing.T) {
+
+	n := Number{}
+	n.SetInt64(42)
+	assert.True(t, n.ntype() == int64Type)
+
+	val, err := n.Int64()
+	assert.True(t, val == 42)
+	assert.True(t, err == nil)
+
+	_, err = n.Float64()
+	assert.True(t, err != nil)
+
+}
+
+// Test_isFloat64 tests if the number is a float64. It should return set value for Float64 and an error for Int64.
+func Test_isFloat64(t *testing.T) {
+
+	n := Number{}
+	n.SetFloat64(42.0)
+	assert.True(t, n.ntype() == float64Type)
+
+	val, err := n.Float64()
+	assert.True(t, val == 42.0)
+	assert.True(t, err == nil)
+
+	_, err = n.Int64()
+	assert.True(t, err != nil)
+
+}
+
+// Test_reset tests resetting the number. It should return invalidType.
+func Test_reset(t *testing.T) {
+
+	n := Number{}
+	n.SetInt64(42)
+	assert.True(t, n.ntype() == int64Type)
+	n.Reset()
+	assert.True(t, n.ntype() == invalidType)
+
+}
+
+// Test_constructor tests the constructor of the number. It should return the correct type.
+func Test_constructor(t *testing.T) {
+
+	n := Number{}
+	assert.True(t, n.ntype() == invalidType)
+
+	n = Number{[9]byte{0, 0, 0, 0, 0, 0, 0, 0, backfillType}}
+	assert.True(t, n.ntype() == backfillType)
+
+	n = Number{[9]byte{0, 0, 0, 0, 0, 0, 0, 0, int64Type}}
+	assert.True(t, n.ntype() == int64Type)
+
+	n = Number{[9]byte{0, 0, 0, 0, 0, 0, 0, 0, float64Type}}
+	assert.True(t, n.ntype() == float64Type)
+
+}
```

Comment:
I am not sure what this test is trying to do? Based on set of functions and given that the `bytes` field is private. I do not think we would initialize the `Number` type in this way.

But also the type checking I think you are already doing this through the `Set**Type` functions. And this initialization to similar to those Set functions.

Reply:
Right makes sense. Removing the constructor initialization